zendesk_tickets/views.py

```diff
@@ -14,11 +23,31 @@ def ticket(request,
     if request.method == 'POST':
         form = form_class(request.POST)
         if form.is_valid():
-            form.submit_ticket(request, subject, tags, ticket_template_name)
-            return HttpResponseRedirect(success_redirect_url)
+            try:
+                form.submit_ticket(request, subject, tags, ticket_template_name)
+                referer = form.cleaned_data.get('referer')
+                if referer:
+                    come_from = urlparse(referer).path
+                    success_redirect_url = (
```

Comment:
i think for a little bit of safety, it might be good to have a try/except wrapper and then an `if come_from:` where you're appending the extra parameter to the url